app/src/main/java/com/example/drawingactivity/LayerRecyclerViewAdapter.kt

```diff
@@ -0,0 +1,84 @@
+package com.example.drawingactivity
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import com.example.drawingactivity.AdapterViewModel.Companion.LAYER
+
+/**
+ * This is a RecyclerView that manages a list and how it is displayed in views
+ */
+class RecyclerViewAdapter(
+    private val data: List<AdapterViewModel>
+) : RecyclerView.Adapter<ViewHolder>() {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+        val inflater = LayoutInflater.from(parent.context)
+        return when (viewType) {
+            LAYER -> LayerViewHolder(inflater.inflate(R.layout.layer_view, parent, false))
+            else -> {
+                throw RuntimeException("View Does Not Exist.")
+            }
+        }
+    }
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        holder.bind(data[position])
+    }
+
+    override fun getItemCount(): Int = data.size
+
+    override fun getItemViewType(position: Int): Int = data[position].type
+
+}
+
+abstract class AdapterViewModel(val type: Int) {
+    companion object {
+        const val LAYER = 0
```

Comment:
again you don't really need types anymore now that there is only one thing that can be in there